src/visualizers/VisualizerHistogram.ts

```diff
@@ -0,0 +1,211 @@
+        }
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        if (this.params.firstIndex.value < 1) {
+            status.isValid = false
+            status.errors.push('First Index can not be less than 1')
+        }
+
+        return status
+    }
+
+    // Find the largest value of the sequence asked for.
+    largestValue(): bigint {
+        let largest_value = 0n
+        for (
+            let i = Math.max(this.firstIndex, this.seq.first);
+            i < Math.min(this.terms, this.seq.last);
+            i++
+        ) {
+            const value = this.seq.getElement(i)
+            if (i == 0) {
+                largest_value = value
+            } else if (value > largest_value) {
+                largest_value = value
+            }
+        }
+        return largest_value
+    }
+
+    // Create an array with the number of factors of
+    // each element at the corresponding index of the array.
+    factorArray(): number[] {
+        const factorArray = []
+        for (
+            let i = Math.max(this.firstIndex, this.seq.first);
+            i < Math.min(this.terms, this.seq.last);
+            i++
+        ) {
+            let counter = 0
+            const factors = this.seq.getFactors(i)
+            if (factors) {
+                for (const [, power] of factors) {
+                    counter += Number(power)
+                }
+            }
+
+            factorArray[i] = counter
+        }
+        return factorArray
+    }
+
+    // Create an array with the frequency of each number
+    // of factors in the corresponding index.
+    orderedFactorArray(): number[] {
+        const orderedFactorArray = []
+        const factorArray = this.factorArray()
+        for (
+            let i = 0;
+            i < natlog(this.largestValue()) / natlog(2) + 1;
+            i++
+        ) {
+            orderedFactorArray.push(0)
+        }
+
+        for (let i = 0; i < factorArray.length; i++) {
+            orderedFactorArray[factorArray[i]]++
+        }
+
+        return orderedFactorArray
+    }
+
+    // Create an array to change the bin size of the
+    // histogram to what the user asked for.
+    binFactorArray(): number[] {
+        const binFactorArray = []
+        const orderedFactorArray = this.orderedFactorArray()
+        for (
+            let i = 0;
+            i < Math.ceil(orderedFactorArray.length / this.binSize);
+            i++
+        ) {
+            binFactorArray.push(0)
+        }
+
+        let index = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            for (let j = 0; j < this.binSize; j++) {
+                if (index + j < orderedFactorArray.length) {
+                    binFactorArray[i] += orderedFactorArray[index + j]
+                }
+            }
+            index += this.binSize
+        }
+
+        return binFactorArray
+    }
+
```

Comment:
Why do you first make orderedFactorArray and then binFactor array? Isn't it clearer and more efficient to just loop over the factorArray once, adding one to the proper bin for each entry? You should be able to compute the bin by dividing and truncating...